flink-dist/src/main/flink-bin/bin/config.sh

```diff
@@ -660,3 +660,39 @@ extractExecutionParams() {
 
     echo ${execution_config} | sed "s/$EXECUTION_PREFIX//"
 }
+
+getFlinkDaemonID() {
+    local pid=$1
+    local clean_up_when_all_processes_dead=${2:-1}
+
+    mkdir -p "$FLINK_PID_DIR"
+
+    # Log files for daemons are indexed from the process ID's position in the PID
+    # file. The following lock prevents a race condition during daemon startup
+    # when multiple daemons read, index, and write to the PID file concurrently.
+    # The lock is created on the PID directory since a lock file cannot be safely
+    # removed. The daemon is started with the lock closed and the lock remains
+    # active in this script until the script exits.
+    command -v flock >/dev/null 2>&1
+    if [[ $? -eq 0 ]]; then
+        exec 200<"$FLINK_PID_DIR"
+        flock 200
+    fi
+
+    # Remove the pid file when all the processes are dead
+    if [ -f "$pid" ];then
+        local all_dead=0
+        while read each_pid; do
+            kill -0 $each_pid > /dev/null 2>&1
+            if [[ $? -eq 0 ]] ; then
+                all_dead=1
+            fi
+        done < "$pid"
+        [ $clean_up_when_all_processes_dead -eq 0 -a $all_dead -eq 0 ] && rm $pid
+    fi
+
+    # Ascending ID depending on number of lines in pid file.
+    # This allows us to start multiple daemon of each type.
+    local id=$([ -f "$pid" ] && echo $(wc -l < "$pid") || echo "0")
+    echo $id
```

Comment:
Can we not simplify this?
```suggestion
    echo $([ -f "$pid" ] && echo $(wc -l < "$pid") || echo "0")
```